### ISSUE TITLE
[Feature Request] Add more granularity to object-literal-shorthand rule options

### ISSUE THREAD

**Issue by UselessPickles** [palantir/tslint#3778]
The `object-literal-shorthand` rule currently does not offer granularity for enforcing/disallowing shorthand separately for property assignment vs method declarations. 

I would prefer to allow (but not enforce!) shorthand for method declarations, but disallow shorthand for property assignments. 

Here's an example of how I would like to configure the `object-literal-shorthand` rule:

```ts
const someValue = 20;

const myObj = { 
    // error: shorthand property assignment not allowed
    someValue,
    // valid
    anotherValue: someValue,
    // valid
    someMethod(): number {
        return 42;
    },
    // also valid
    anotherMethod: (): number => {
       return 42;
    }
};
```

The motivation for this is that shorthand method declaration is very convenient and improves readability, while shorthand property assignment creates a coupling between the object's property name and a local variable name that can lead to bugs when refactoring code later.

**Comment by JoshuaKGoldberg** [comment]
Note: per #4534, this issue will be closed in less than a month if no PR is sent to add the rule. If you _really_ need the rule, custom rules are always an option and can be maintained outside this repo!

**Comment by pablobirukov** [comment]
Reopened https://github.com/palantir/tslint/pull/4139 would resolve the issue, if you consider releasing a new version

**Comment by JoshuaKGoldberg** [comment]
Oh hey, this actually won't be closed, as it's a feature request and not a rule suggestion. Whoops!

And yes, absolutely! If you're up for reopening _(or recreating)_ #4139, we'd be happy to review it.
